tests/comptroller/utils.ts

```diff
@@ -0,0 +1,173 @@
+
+  newMarketListed.parameters.push(mTokenParam);
+  return newMarketListed;
+}
+
+export function handleMarketListeds(events: MarketListed[]): void {
+  events.forEach(event => {
+    handleMarketListed(event)
+  })
+}
+
+export function createNewMintEvent(from: string, to: string, amount: BigInt, hash: string): Mint {
+  let minterParam = new ethereum.EventParam("minter", ethereum.Value.fromAddress(Address.fromString(from)));
+  let mintAmountParam = new ethereum.EventParam("mintAmount", ethereum.Value.fromUnsignedBigInt(amount));
+  let mintTokensParam = new ethereum.EventParam("mintTokens", ethereum.Value.fromUnsignedBigInt(amount));
+
+  let newMint = changetype<Mint>(newMockEventWithParams([minterParam, mintAmountParam, mintTokensParam]));
+  newMint.address = Address.fromString(to); // Set the address of the event to the mToken address
+  newMint.transaction.hash = Bytes.fromHexString(hash) as Bytes; // Set the hash of the transaction
+  return newMint;
+}
+
+export function handleNewMints(events: Mint[]): void {
+  events.forEach(event => {
+    handleMint(event)
+  })
+}
+
+export function createNewTransferEvent(market: string, from: string, to: string, amount: BigInt, hash: string): Transfer {
+  let fromParam = new ethereum.EventParam("from", ethereum.Value.fromAddress(Address.fromString(from)));
+  let toParam = new ethereum.EventParam("to", ethereum.Value.fromAddress(Address.fromString(to)));
+  let amountParam = new ethereum.EventParam("amount", ethereum.Value.fromUnsignedBigInt(amount));
+
+  let newTransfer = changetype<Transfer>(newMockEventWithParams([fromParam, toParam, amountParam]));
+  newTransfer.address = Address.fromString(market); // Set the address of the event to the mToken address
+  newTransfer.transaction.hash = Bytes.fromHexString(hash) as Bytes; // Set the hash of the transaction
+  return newTransfer;
+}
+
+export function handleNewTransfers(events: Transfer[]): void {
+  events.forEach(event => {
+    handleTransfer(event)
+  })
+}
+
+export function createNewRedeemEvent(from: string, to: string, amount: BigInt, hash: string): Redeem {
+  let redeemerParam = new ethereum.EventParam("redeemer", ethereum.Value.fromAddress(Address.fromString(from)));
+  let redeemAmountParam = new ethereum.EventParam("redeemAmount", ethereum.Value.fromUnsignedBigInt(amount));
+  let redeemTokensParam = new ethereum.EventParam("redeemTokens", ethereum.Value.fromUnsignedBigInt(amount));
+
+  let newRedeem = changetype<Redeem>(newMockEventWithParams([redeemerParam, redeemAmountParam, redeemTokensParam]));
+  newRedeem.address = Address.fromString(to); // Set the address of the event to the mToken address
+  newRedeem.transaction.hash = Bytes.fromHexString(hash) as Bytes; // Set the hash of the transaction
+  return newRedeem;
+}
+
+export function handleNewRedeems(events: Redeem[]): void {
+  events.forEach(event => {
+    handleRedeem(event)
```

Comment:
We're definitely covering `handleRedeem` - not sure what is up with code coverage but I'll troubleshoot more.